packages/tenant-process/src/services/tenantService.ts

```diff
@@ -71,49 +74,78 @@ export function tenantServiceBuilder(
       );
     },
 
-    async updateTenantAttribute(
-      tenantId: string,
-      attributeId: string,
-      newAttribute: TenantAttribute
-    ): Promise<string> {
+    async updateTenantVerifiedAttribute({
+      verifierId,
+      tenantId,
+      attributeId,
+      updateVerifiedTenantAttributeSeed,
+    }: {
+      verifierId: string;
+      tenantId: string;
+      attributeId: string;
+      updateVerifiedTenantAttributeSeed: UpdateVerifiedTenantAttributeSeed;
+    }): Promise<void> {
       const tenant = await readModelService.getTenantById(tenantId);
 
-      return await repository.createEvent(
-        await updateTenantAttributeLogic({
+      await repository.createEvent(
+        await updateTenantVerifiedAttributeLogic({
+          verifierId,
           tenant,
           tenantId,
           attributeId,
-          newAttribute,
+          updateVerifiedTenantAttributeSeed,
         })
       );
     },
   };
 }
-export async function updateTenantAttributeLogic({
+
+async function updateTenantVerifiedAttributeLogic({
+  verifierId,
   tenant,
   tenantId,
   attributeId,
-  newAttribute,
+  updateVerifiedTenantAttributeSeed,
 }: {
+  verifierId: string;
   tenant: WithMetadata<Tenant> | undefined;
   tenantId: string;
   attributeId: string;
-  newAttribute: TenantAttribute;
+  updateVerifiedTenantAttributeSeed: UpdateVerifiedTenantAttributeSeed;
 }): Promise<CreateEvent<TenantEvent>> {
   assertTenantExists(tenantId, tenant);
-  if (!newAttribute || newAttribute.id !== attributeId) {
-    throw invalidAttributeStructure;
-  }
-  assertAttributeExists(attributeId, tenant.data.attributes);
 
-  const updatedAttributes = [
-    newAttribute,
-    ...tenant.data.attributes.filter((a) => a.id !== newAttribute.id),
-  ];
+  const expirationDate = updateVerifiedTenantAttributeSeed.expirationDate
+    ? new Date(updateVerifiedTenantAttributeSeed.expirationDate)
+    : undefined;
+
+  assertValidExpirationDate(expirationDate);
+
+  const attribute = tenant?.data.attributes.find(
+    (att) => att.id === attributeId
+  );
```

Comment:
```suggestion
  const attribute = tenant?.data.attributes
    .filter((att) => att.type === tenantAttributeType.VERIFIED)
    .find((att) => att.id === attributeId);
```

Maybe we can move this in the `assertVerifiedAttributeExistsInTenant` and adjust its logic?

Reply:
I don't understand what you mean, sorry